lib/solveEquation/stepThrough.js

```diff
@@ -107,6 +132,48 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   return steps;
 }
 
+/*
+  Helper function that returns the roots and symbol of an input equation
+  that has roots to be found.
+  For every factor on the left hand side, solve a new equation that is factor = 0
+  for the symbol.
+  TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
+*/
+
+function getSolutionsAndSymbol (equation) {
+  // Gets all factors that aren't a constant
+  const factors = equation.leftNode.args.filter(arg => !checks.resolvesToConstant(arg));
+
+  const solutions = [];
+
+  let symbol;
+
+  // For each factor, solve the equation factor = 0 for the symbol
+  for (var f in factors) {
+    const leftNode = Node.Type.isParenthesis(factors[f])
+          ? factors[f].content
+          : factors[f];
+
+    const rightNode = Node.Creator.constant(0);
+
+    const allSteps = stepThrough(leftNode, rightNode, '=');
+
+    if (allSteps.length > 0) {
+      const lastStep = allSteps.slice(-1)[0];
+
+      symbol = lastStep.newEquation.leftNode;
```

Comment:
You still haven't addressed that this might not be a symbol. The equation code is far from perfect and finished, and so there can be steps without ending up with `symbol = solution` at the end (at least I think so - if you're pretty sure that's wrong feel free to show me the code that guarantees that!)

 I think you should explicitly check that and if it's not a symbol, don't push to `solutions`